src/server.c

```diff
@@ -3480,8 +3480,13 @@ void call(client *c, int flags) {
         char *latency_event = (c->cmd->flags & CMD_FAST) ?
                               "fast-command" : "command";
         latencyAddSampleIfNeeded(latency_event,duration/1000);
-        slowlogPushEntryIfNeeded(c,c->argv,c->argc,duration);
+        /* If command argument vector was rewritten, use the original
+         * arguments. */
+        robj **argv = c->original_argv ? c->original_argv : c->argv;
+        int argc = c->original_argv ? c->original_argc : c->argc;
+        slowlogPushEntryIfNeeded(c,argv,argc,duration);
     }
+    freeClientOriginalArgv(c);
```

Comment:
maybe this better be moved into `freeClientArgv`?

Reply:
This doesn't work in replaceClientCommandVector since we both free the original arguments and retain them.

Reply:
It also interacts a little weirdly with multi-exec, since freeClientArgv isn't called in the multi flow, so it would require an explicit call there. That might be worth refactoring, but would rather split this effort from that.